treeherder/etl/resultset_loader.py

```diff
@@ -182,5 +184,75 @@ def transform(self, repository):
         return self.fetch_resultset(pr_url, repository)
 
 
+class HgPushTransformer:
+    # {
+    #   "root": {
+    #     "payload": {
+    #       "pushlog_pushes": [
+    #         {
+    #           "time": 14698302460,
+    #           "push_full_json_url": "https://hg.mozilla.org/try/json-pushes?version=2&full=1&startID=136597&endID=136598",
+    #           "pushid": 136598,
+    #           "push_json_url": " https: //hg.mozilla.org/try/json-pushes?version=2&startID=136597&endID=136598",
+    #           "user": " james@hoppipolla.co.uk"
+    #         }
+    #       ],
+    #       "heads": [
+    #         "2f77bc4f354d9ba67ea5270b2fc789f4b0521287"
+    #       ],
+    #       "repo_url": "https://hg.mozilla.org/try",
+    #       "_meta": {
+    #         "sent": "2016-07-29T22:11:18.503365",
+    #         "routing_key": "try",
+    #         "serializer": "json",
+    #         "exchange": "exchange/hgpushes/v1"
+    #       }
+    #     }
+    #   }
+    # }
+
+    def __init__(self, message_body):
+        self.message_body = message_body
+        self.repo_url = message_body["payload"]["repo_url"]
+        self.branch = None
+
+    def transform(self, repository):
+        logger.info("transforming for {}".format(repository))
+        url = self.message_body["payload"]["pushlog_pushes"][0]["push_full_json_url"]
+        return self.fetch_resultset(url, repository)
+
+    def fetch_resultset(self, url, repository, sha=None):
+        try:
+            logger.info("fetching for {} {}".format(repository, url))
+            # there will only ever be one, with this url
+            push = fetch_json(url)["pushes"].values()[0]
+            commits = list(push["changesets"])
+            head_commit = commits[0]
+            logger.info("got head commit: {}".format(head_commit["node"]))
+            resultset = {
```

Comment:
This doesn't set `revision_hash` like pushlog.py does. I know we want to get rid of it as soon as we can, but I think we're not going to be able to do so quite yet sadly.